src/app/pages/google-email/google-email.component.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, OnInit } from '@angular/core';
+import { UserInfo } from 'angular-oauth2-oidc';
+import { GoogleApiService } from '../../services/google-api.service';
+import { async, lastValueFrom } from 'rxjs';
+import { DeleteConfirmationComponent } from '../../components/delete-confirmation/delete-confirmation.component';
+import { MatDialog } from '@angular/material/dialog';
+
+@Component({
+  selector: 'app-google-email',
+  templateUrl: './google-email.component.html',
+  styleUrls: ['./google-email.component.scss'],
+})
+export class GoogleEmailComponent implements OnInit {
+  mailDetails: {
+    labels: any;
+    date: Date;
+    id: string;
+    sender: string;
+    snippet: string;
+    body: string;
+  }[] = [];
+  userInfo?: UserInfo;
+  loadingEmails = false;
+  loading = false;
+
+  constructor(private readonly googleApi: GoogleApiService, public dialog: MatDialog) {}
+
+  ngOnInit() {
+    const user = localStorage.getItem('user');
+    if (user) {
+      this.userInfo = JSON.parse(user);
+    }
+    this.getEmails().then((r) => console.log(r));
+  }
+
+  async confirmDelete(mailDetail: any) {
```

Comment:
what is a mailDetail composed of ? if you can catch it you can replace any type by a mailDetail Model or interface

Reply:
done